ungrib/src/rrpr.F

```diff
@@ -1451,3 +1460,56 @@ subroutine fix_gefs_landmask (ix, jx, plvl)
     deallocate (sea)
   endif
 end subroutine fix_gefs_landmask
+
+subroutine fix_icon_soilm (ix, jx)
+  use storage_module
+  implicit none
+  integer :: ix, jx
+  real, allocatable, dimension(:,:) :: soilm000, soilm002, soilm006, &
+                     soilm018, soilm054, soilm162, soilm486, soilm999
+  allocate(soilm000(ix,jx))
+  allocate(soilm002(ix,jx))
+  allocate(soilm006(ix,jx))
+  allocate(soilm018(ix,jx))
+  allocate(soilm054(ix,jx))
+  allocate(soilm162(ix,jx))
+  allocate(soilm486(ix,jx))
+  allocate(soilm999(ix,jx))
+  call get_storage(200100, 'SOILM000', soilm000, ix, jx)
+  call get_storage(200100, 'SOILM002', soilm002, ix, jx)
+  call get_storage(200100, 'SOILM006', soilm006, ix, jx)
+  call get_storage(200100, 'SOILM018', soilm018, ix, jx)
+  call get_storage(200100, 'SOILM054', soilm054, ix, jx)
+  call get_storage(200100, 'SOILM162', soilm162, ix, jx)
+  call get_storage(200100, 'SOILM486', soilm486, ix, jx)
+  call get_storage(200100, 'SOILM999', soilm999, ix, jx)
+    !convert kg m-2 to m3 m-3
+    soilm001 = soilm000 / 10.
```

Comment:
@WQadam - Did you mean to use `soilm001` here? I imagine you meant to use `soilm000`?